xds/internal/resolver/xds_resolver_test.go

```diff
@@ -860,6 +850,8 @@ func (s) TestXDSResolverDelayedOnCommitted(t *testing.T) {
 			},
 		},
 	}, nil)
+	tcc.stateCh.Receive(ctx) // Ignore the first update
```

Comment:
Move comment to above line of code (Easwar mentioned he doesn't like it beside.) Add period to comment. Why do we need to invoke the callback twice? None of the "handling" logic (inline instead of explicit functions like in cds) in xds resolver run() leads me to believe we need to invoke it twice.

Reply:
> Move comment to above line of code (Easwar mentioned he doesn't like it beside.)

This is pretty common practice in my experience.  @easwars - is there some justification behind what otherwise sounds like personal preference?  (FWIW I urge everyone to push back when someone - including me - suggests something in a review that sounds subjective; there may be objective benefits you aren't seeing, but not necessarily.  And if there are objective benefits but you are not understanding them, then everyone loses.)

> Add period to comment.

I'll give you this one. :)  The rule I've always heard and followed is: comments that are not sentences are fine (`count int // the number of things`), but then you don't capitalize the first letter or end with a period.  Putting a period at the end of a sentence fragment is not proper.  This one turns out to (technically) be a legal complete sentence due to the implied "you" as a command, though I didn't think of it that way initially.

> Why do we need to invoke the callback twice?

In first update after any cluster removal, the previous config selector is still live and references the cluster, meaning it needs to remain live in the system.  With the second update, any _otherwise-unreferenced_ clusters may be removed.  In this case, the initially-selected cluster is still live, since we haven't "finished" the "RPC" yet, so it _should_ not be removed from the list of live clusters; this test confirms that.

There is a comment: "Perform TWO updates to ensure the old config selector does not hold a reference to test-cluster-1" - LMK if this isn't clear enough given this explanation of how config selectors & cluster removal works.

Reply:
> Perhaps you could add your explanation to me to the comment in the test

I feel like the comment that is already there is sufficient given an understanding of how the config selector / cluster stuff works at a semi-high level.  This is actually discussed in the design of the feature -- which I know you didn't read, and I wouldn't expect you to for one small PR.  That kind of understanding should be something test writers can assume anyone reading the test already has.

If you think something is not clear given "reasonable" background (here or in a future PR), then suggesting exactly how to add a comment or change an existing one can be helpful.

Thanks!

Reply:
> Move comment to above line of code (Easwar mentioned he doesn't like it beside.)

Wasn't able to find where I said I don't like comments which are beside the line of code. I might have asked you to terminate the comment with a punctuation. 

> Easwar is the community contributor about go style so I took his word as gospel

Lol ... you don't have to. You could have a slightly different interpretation of the style guide and I'm also allowed to be wrong from time to time :)